foundry/applications/formApplications/baseEntitySheets/journalSheet.d.ts

```diff
@@ -1,23 +1,67 @@
 /**
  * The JournalEntry Configuration Sheet
- *
- * @param entity - The JournalEntry instance which is being edited
- * @param options - Application options
  */
-declare class JournalSheet extends BaseEntitySheet {
+declare class JournalSheet<
+  D extends object = BaseEntitySheet.Data<JournalEntry>,
+  O extends JournalEntry = D extends BaseEntitySheet.Data<infer T> ? T : JournalEntry
+> extends BaseEntitySheet<D, O> {
+  constructor(object: O, options?: JournalSheet.Options);
+
+  /** @override */
+  static get defaultOptions(): JournalSheet.Options;
+
+  /** @override */
+  get id(): string;
+
+  /** @override */
+  get template(): string;
+
+  /** @override */
+  get title(): string;
+
+  /** @override */
+  getData(options?: Application.RenderOptions): Promise<D> | D;
+
   /**
-   * Provide a unique CSS ID for Entity Sheets
+   * Guess the default view mode for the sheet based on the player's permissions to the Entry
    */
-  get id(): string;
+  protected _inferDefaultMode(): JournalSheet.SheetMode;
+
+  /** @override */
+  protected _render(force?: boolean, options?: JournalSheet.RenderOptions): Promise<void>;
+
+  /** @override */
+  protected _getHeaderButtons(): Application.HeaderButton[];
+
+  /** @override */
+  protected _updateObject(event: Event, formData: object): Promise<O>;
 
   /**
-   * Suppress the JournalEntry title when an image is shown to players whom do not have at least limited permission
+   * Handle requests to switch the rendered mode of the Journal Entry sheet
+   * Save the form before triggering the show request, in case content has changed
    */
-  get title(): string;
+  protected _onSwapMode(event: Event, mode: JournalSheet.SheetMode): Promise<void>;
 
   /**
-   * @param options - (unused)
-   * @override
+   * Handle requests to show the referenced Journal Entry to other Users
+   * Save the form before triggering the show request, in case content has changed
    */
-  getData(options?: any): BaseEntitySheet.Data<Entity>;
+  protected _onShowPlayers(event: Event): Promise<void>;
+}
+
+declare namespace JournalSheet {
+  interface RenderOptions extends Application.RenderOptions {
+    sheetMode: SheetMode;
+  }
+
+  interface Options extends BaseEntitySheet.Options {
+    sheetMode: SheetMode;
```

Comment:
It should also allow `null` because the class handles `null` properly (it can be returned by `_inferDefaultMode` anyways).

Reply:
You're right, I guess it can only be undefined as well? So it should be `sheetMode?: SheetMode | null`?

Reply:
Yeah, I suppose. :+1: